unyt/array.py

```diff
@@ -339,6 +339,67 @@ def _apply_power_mapping(ufunc, in_unit, in_size, in_shape, input_kwarg_dict):
     return mul, unit
 
 
+def _subclass_ufunc_prepare_and_finalize(ufunc_handler):
+    """
+    This wrapper function is intended to be applied to __array_ufunc__ on
+    unyt_array.
+
+    It enables subclasses of unyt_array and unyt_quantity to define two
+    classmethods: __unyt_ufunc_prepare__ and __unyt_ufunc_finalize__.
+    These are called if available to allow subclasses to prepare arguments
+    for and modify return values from numpy ufuncs.
+
+    The __unyt_ufunc_prepare__ method should be a classmethod accepting
+    (ufunc, method, *inputs, **kwargs), i.e. the same inputs as
+    __array_ufunc__. It should return
+    (prepared_ufunc, prepared_method, prepared_inputs, prepared_kwargs)
+    which correspond logically to its inputs with any modifications needed
+    by the subclass.
+
+    The __unyt_ufunc_finalize__ method should be a classmethod accepting
+    (result, ufunc, method, *inputs, **kwargs), similarly to above but with
+    the addition of result, the return value from unyt_array.__array_ufunc__.
+    It should return the result with any modifications needed by the subclass.
+    """
+
+    def wrapper(self, ufunc, method, *inputs, **kwargs):
+        if len(inputs) > 1:
```

Comment:
Some comments in these lines as to what is being attempted here and what might raise `RuntimeError` would be helpful.

Reply:
Added some comments here.